contracts/collections/sg721-base/src/lib.rs

```diff
@@ -24,8 +25,9 @@ pub mod entry {
     use sg_std::Response;
 
     // version info for migration info
-    const CONTRACT_NAME: &str = "crates.io:sg721-base";
-    const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+    pub const CONTRACT_NAME: &str = "crates.io:sg721-base";
+    pub const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+    pub const EXPECTED_FROM_VERSION: &str = "0.16.0";
```

Comment:
I don't think old versions should be hardcoded. Extra work for developers and more things to remember.

Reply:
Note that I am following the logic from the cw 0.17 upgrade here: 
https://github.com/CosmWasm/cw-nfts/pull/112/files#diff-733b03d19162a7149fd3e3feb6552118ea6a667aac4cbcda9ece681ad6425ef7R39

Please refer to this for all of the 0.16 comments. It wouldn't make sense for me to do the update logic in a different way than the underlying cw-nfts logic that I am copying.

Reply:
The other way to do this is a set of whitelisted versions. See sg721-updatable/contract.rs migrate fn

Reply:
@yubrew can you clarify? I don't see the word "migrate" in this file: 

https://github.com/public-awesome/launchpad/blob/main/contracts/collections/sg721-updatable/src/contract.rs

Reply:
> I don't think old versions should be hardcoded.

@shanev I'm considering organizing upgrade logics like this:

```plain
.
└── src/
    └── upgrades/
        ├── mod.rs
        ├── v0_17.rs
        ├── v0_18.rs
        └── v0_19.rs
```

`v0_17.rs` contains the migration logics from 0.16 to 0.17; `v0_18.rs` contains logics for 0.17 => 0.18, so on.

In 0.16 => 0.17 we may need to assert that the contract being upgraded is at 0.16. This may be because older versions (e.g. 0.15) may have different layouts that are incompatible with the migration logics. In this case, imo it's reasonable to hardcode a version.

Similarly if 0.17 => 0.18 migration logics only work on 0.17, it may also hardcode an expected version accordingly.